src/main/java/org/jenkinsci/plugins/stashNotifier/StashNotifier.java

```diff
@@ -226,14 +236,20 @@ private boolean perform(Run<?, ?> run,
 
 		PrintStream logger = listener.getLogger();
 
-		Result result = run.getResult();
-		if (result == null && disableInProgress) {
+		Result buildResult = run.getResult();
+		if (buildResult == null && disableInProgress) {
 			return true;
-		} else if (result == null) {
+		} else if (buildResult == null) {
 			state = StashBuildState.INPROGRESS;
-		} else if (result.equals(Result.SUCCESS)) {
+		} else if (buildResult == Result.SUCCESS) {
 			state = StashBuildState.SUCCESSFUL;
-		} else if (result.equals(Result.NOT_BUILT)) {
+		} else if (buildResult == Result.UNSTABLE && considerUnstableAsSuccess) {
+			logger.println("UNSTABLE reported to stash as SUCCESSFUL");
+			state = StashBuildState.SUCCESSFUL;
+		} else if (buildResult == Result.ABORTED && (disableAbortAsFailures || disableInProgress)) {
```

Comment:
Hold on, if we disable the in progress *and* the build has just being canceled, why should we mark it as failed after all?
On the other side you are right, if we started saying that the build is in progress *and* has been canceled, we have to tell Stash somehow.

Does this mean that we do not need any disableAbortAsFailure *and* we just notify the abort only if we informed Stash that the build was in progress?